bayesian/utility/db_gateway.py

```diff
@@ -63,3 +73,54 @@ def get_ca_data_from_graph(cls, ecosystem, package, version, vendor):
         elapsed_seconds = (datetime.now() - start).total_seconds()
         logger.info("Gremlin request took {} seconds.".format(elapsed_seconds))
         return query_result.json()
+
+
+class RdbAnalyses:
+    """RDB Interface class used by stack analyses flow.
+
+    Provides interfaces to save and read request post data for stack analyses v2.
+    """
+
+    def __init__(self, request_id, submit_time=None, manifest=None, deps=None):
+        """Set request id."""
+        self.request_id = request_id
+        self.submit_time = submit_time
+        self.manifest = manifest
+        self.deps = deps
+
+    def get_request_data(self):
+        """Read request data for given request id from RDS."""
+        db_result = fetch_sa_request(rdb, self.request_id)
+        if db_result is None:
+            error_message = 'Invalid request ID {}.'.format(self.request_id)
+            logger.exception(error_message)
+            raise HTTPError(404, error_message)
```

Comment:
We should not be raising HTTPErrors from this layer. Please move to API layer

Reply:
@prashbnair , I was the one who suggested to use exceptions instead of lots of nested if..else. Do you want him to use custom exception classes instead of throwing generic HttpError from business layer or you don't want to use exceptions at all? 

Unlike any other programming languages, using exceptions are prevalent in Python and even Python core developers recommends to use for flow control. P.S. https://docs.python.org/2/glossary.html#term-eafp